app/models/user.php

```diff
@@ -0,0 +1,80 @@
+<?php
+
+class User extends AppModel
+{
+    private $is_login_valid = true;
+
+    public $validation = array(
+
+            'username' => array(
+                'duplicate' => array('isRegistered'
+                ),
+                'length' => array(
+                    'validate_between', MIN_USERNAME_CHARACTERS, MAX_USERNAME_CHARACTERS,
+                ),
+            ),
+
+            'password' => array(
+                'length' => array(
+                    'validate_between', MIN_PASSWORD_CHARACTERS, MAX_PASSWORD_CHARACTERS,
+                ),
+            ),
+    );
+
+    private function getRowByUsername()
+    {
+        $db = DB::conn();
+        $query = "SELECT id FROM user WHERE BINARY username = ?";
+        $row = $db->row($query, array($this->username));
+
+        return $row;
+    }
+
+    //Check if the username is already registered
+    public function isRegistered()
+    {
+        $row = self::getRowByUsername();
+        return(empty($row));
+    }
+
+    //Adds a user
+    public function register()
+    {
+        if (!$this->validate()) {
+            throw new ValidationException('Invalid username or password input');
+        }
+
+        $input = array(
+            'username' => $this->username,
+            'password' => $this->password
+        );
+
+        $row = self::getRowByUsername();
+        if ($row) {
```

Comment:
you can directly use the method po if it returns boolean to save one variable declaration :bow:
```php
if (self::getRowByUsername()) {
...
}
```